frontend/apps/reader/modules/readerview.lua

```diff
@@ -792,7 +792,19 @@ function ReaderView:onReadSettings(config)
     self.highlight.saved = config:readSetting("highlight", {})
     -- Highlight formats in crengine and mupdf are incompatible.
     -- Backup highlights when the document is opened with incompatible engine.
-    local _, page_highlights = next(self.highlight.saved) -- get the first page with highlights
+    local page, page_highlights
+    while true do -- remove empty tables for pages without highlights and get the first page with highlights
+        page, page_highlights = next(self.highlight.saved, page)
+        if page then
+            if #page_highlights == 0 then
+                self.highlight.saved[page] = nil
```

Comment:
I'm not sure this will work well, `next()` and removing the current one: the next `next()` will get as the 2nd arg a key that doesn't exist anymore, dunno if it will reset from start or from end. Pinging @NiLuJe if you have any thought on this behaviour.
May be explicitely resetting also `page = nil` is safer, so it starts as the initial call. Or just don't provide any 2nd arg so we always restart from 0, as you break as soon as you have one.

Reply:
> so we always restart from 0

The safest way, thank you.